### ISSUE TITLE
Add role to search results sort by button

### ISSUE THREAD

**Issue by atsansone** [flutter/website#10542]
From https://buganizer.corp.google.com/issues/337311368

### Observed Results

Navigating through the page sort by button, listen that it is announced as a text without a role button.

* [Screencast VoiceOver](http://screencast/cast/NTM5MDc2MjIxOTkyOTYwMHxhMDIyY2QwMi04ZQ)
* [Screencast ChromeVox](http://screencast/cast/NjI3ODUyNjI1NTc1OTM2MHxkMjJkYzk3My0wNw)

### Expected Results

Ensure that when the user is navigating the page and selects the sort by button it should have the correct role to the user in this case is a button which needs to be announced also the state.

Sort by button collapsed/expanded

### User Impact

Screen reader and Cognitive users are affected because they listen that it is a text and is a button that has options inside.

### Steps to Reproduce

1. Enabled VoiceOver using <kbd>Command</kbd> + <kbd>F5</kbd> also ChromeVox using "Ctrl+Alt+Z'
1. Open the following page: https://docs.flutter.dev/search?q=devtools
1. Continue the linear navigation using the <kbd>Tab</kbd> key.
1. Reach the sort by button and listen the results in SR.

### GAR Assessment

[Flutter Docs](https://a11ytracker.corp.google.com/products/c2fdcf18-53d2-475d-b8b0-e5dedf8cb900?e=ProductLabels)

### GAR Details

[1.1. Do all non-text content and user interface components have meaningful accessible names and alternative text?](go/GAR-web#accessible-names)

### Environment Details

Machine: MAC/ChromeBook
Browser: Chrome Version 120.0.6099.272 (Official Build) (64-bit)
Screen reader: VoiceOver / ChromeVox.
Bug filling time: 15 mins.

**Comment by parlough** [comment]
The search experience and UI is implemented with the [Programmable/custom search engine](https://developers.google.com/custom-search). It generates this button, the results, etc. So outside of overriding some styling, I don't think there's much we should do on the client side to change the output.

Perhaps something can be done in the control panel to fix this and other similar issues? I'm not sure as I don't have access, but I also wonder if the team that maintains programmable search engine is tracking these issues or fixing them? Perhaps it's something best left to them as I imagine it would affect all users of the tool.

**Comment by parlough** [comment]
These search results are generated, styled, and managed by the Google Custom Search setup we use.

Since we don't control that, I'm going to close this in favor of an umbrella issue https://github.com/flutter/website/issues/11800 which tracks implementing a new experience that we can properly integrate and style. Thanks!
